File: BE/src/main/java/com/team26/todolist/domain/CardStatus.java
package com.team26.todolist.domain;

import java.util.Arrays;

public enum CardStatus {
    TODO("해야할 일"),
    ONGOING("하고 있는 일"),
    DONE("완료한 일"),
    UNCLASSIFIED("미분류");

    private String name;

    CardStatus(String name) {
        this.name = name;
    }

    //TODO
    public static CardStatus createCardStatus(String cardStatus) {
        return Arrays.stream(CardStatus.values())
                .filter(status -> status.getName().equals(cardStatus))
                .findAny()
                .orElse(CardStatus.UNCLASSIFIED);
    }

    public String getName() {
        return name;
    }
}

File: TodoList/TodoList/domain/entity/CardStatus.swift
//
//  CardStatus.swift
//  TodoList
//
//  Created by Bibi on 2022/04/07.
//

import Foundation

enum CardStatus: Codable {
    case todo
    case doing
    case done

    var name: String {
        switch self {
        case .todo:
            return "해야 할 일"
        case .doing:
            return "하고 있는 일"
        case .done:
            return "완료한 일"
        }
    }
}
